app/channel.go

```diff
@@ -209,6 +209,18 @@ func (a *App) CreateChannelWithUser(c *request.Context, channel *model.Channel,
 	message := model.NewWebSocketEvent(model.WebsocketEventChannelCreated, "", "", userID, nil)
 	message.Add("channel_id", channel.Id)
 	message.Add("team_id", channel.TeamId)
+
+	if category != nil {
+		category.Channels = append([]string{channel.Id}, category.Channels...)
+		var categories []*model.SidebarCategoryWithChannels
+		categories = append(categories, category)
+		_, _, err := a.Srv().Store.Channel().UpdateSidebarCategories(user.Id, channel.TeamId, categories)
+		if err != nil {
+			return nil, model.NewAppError("UpdateSidebarCategories", "app.channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+		}
+		message.Add("categoryId", category.Id)
```

Comment:
Traditionally, we have used `snake_case` for websocket msg properties. Let's continue that.